add --state option to the result method example (#3413)
fishtown-analytics/docs.getdbt.com: website/docs/reference/node-selection/methods.md

### The "result" method
<Changelog>New in v1.0.0</Changelog>

The `result` method is related to the `state` method described above and can be used to select resources based on their result status from a prior run. Note that one of the dbt commands [`run`, `test`, `build`, `seed`] must have been performed in order to create the result on which a result selector operates. You can use `result` selectors in conjunction with the `+` operator. 

```bash
$ dbt run --select result:error --state path/to/artifacts # run all models that generated errors on the prior invocation of dbt run
$ dbt test --select result:fail --state path/to/artifacts # run all tests that failed on the prior invocation of dbt test
$ dbt build --select 1+result:fail --state path/to/artifacts # run all the models associated with failed tests from the prior invocation of dbt build
$ dbt seed --select result:error --state path/to/artifacts # run all seeds that generated errors on the prior invocation of dbt seed.
```

> 

Reviewer: ```suggestion
**Note** &mdash; When you use dbt Cloud, please be aware that the `--state` parameter is automatically included when you click the [**Defer**](/docs/deploy/cloud-ci-job#deferral-and-state-comparison) button.
```

### The "result" method
<Changelog>New in v1.0.0</Changelog>

The `result` method is related to the `state` method described above and can be used to select resources based on their result status from a prior run. Note that one of the dbt commands [`run`, `test`, `build`, `seed`] must have been performed in order to create the result on which a result selector operates. You can use `result` selectors in conjunction with the `+` operator. 

```bash
$ dbt run --select result:error --state path/to/artifacts # run all models that generated errors on the prior invocation of dbt run
$ dbt test --select result:fail --state path/to/artifacts # run all tests that failed on the prior invocation of dbt test
$ dbt build --select 1+result:fail --state path/to/artifacts # run all the models associated with failed tests from the prior invocation of dbt build
$ dbt seed --select result:error --state path/to/artifacts # run all seeds that generated errors on the prior invocation of dbt seed.
```

> **Note** &mdash; When you use dbt Cloud, please be aware that the `--state` parameter is automatically included when you click the [**Defer**](/docs/deploy/cloud-ci-job#deferral-and-state-comparison) button.

Reviewer 2: ```suggestion
**Note** &mdash; When you define steps in your dbt Cloud jobs, please be aware that the `--state` parameter is automatically included when you click the [**Defer**](/docs/deploy/cloud-ci-job#deferral-and-state-comparison) button.
```

Reviewer 2: The only slight clarification is that `--state` is automatically added for **job steps** when the deferral box is ticked.

In the IDE for example people would still have to type `--state ...`

> The `result` method is related to the `state` method described above, and can be used to select resources based on their result status from a prior run. Note that one of the dbt commands [`run`, `test`, `build`, `seed`] must have been performed in order to create the result on which a result selector operates. You can use `result` selectors in conjunction with the `+` operator.

Reviewer: ```suggestion
The `result` method is related to the `state` method described above and can be used to select resources based on their result status from a prior run. Note that one of the dbt commands [`run`, `test`, `build`, `seed`] must have been performed in order to create the result on which a result selector operates. You can use `result` selectors in conjunction with the `+` operator. 
```
